### ISSUE TITLE
How to test trained model without creating memory pool?

### ISSUE THREAD

**Issue by Vancasola** [tensorforce/tensorforce#573]
Hi,
  after training my model, I want to test it without training. I wish the memory usage will be lower t than it when training. 
  I restore the agent and use the flag 'testing', but the model still consumes too much memory usage. And I find that 'testing' only stop updating the network, but the memory pool created when training still exists and take up too much memory.
  So how can I run the trained agent without updating network and creating memory pool? **I wish only the agent and network are load, to consume memory usage as less as possible and to run fast as possible.** 

this is my configure file
` {
    "type": "dqn_agent",

    "update_mode": {
        "unit": "timesteps",
        "batch_size": 100,
        "frequency": 10
    },
    "memory": {
        "type": "replay",
        "capacity": 100000,
        "include_next_states": true
    },

    "optimizer": {
        "type": "clipped_step",
        "clipping_value": 0.1,
        "optimizer": {
            "type": "adam",
            "learning_rate": 1e-3
        }
    },

    "discount": 0.95,
    "entropy_regularization": null,
    "double_q_model": true,

    "target_sync_frequency": 1000,
    "target_update_weight": 1.0,

    "actions_exploration": {
        "type": "epsilon_anneal",
        "initial_epsilon": 1.0,
        "final_epsilon": 0.1,
        "timesteps": 20000
    },

    "saver": {
        "directory":"home/saver",
        "seconds": 60
    },
    "summarizer": {
        "directory": null,
        "labels": ["graph", "total-loss"]
    },
    "execution": {
        "type": "single",
        "session_config": null,
        "distributed_spec": null
    }
}
`

**Comment by AlexKuhnle** [comment]
Hi,
unfortunately this is not easily possible right now. Basically, it shouldn't be too problematic to identify the act-related operations in the graph and then extract only them as TensorFlow model, but there is currently no high-level Tensorforce function which does that, so you would need to do it yourself. I'm happy to help if you want to be in touch e.g. via Gitter, as I would be keen to add this as a general feature to the framework.

**Comment by Vancasola** [comment]
> Hi,
> unfortunately this is not easily possible right now. Basically, it shouldn't be too problematic to identify the act-related operations in the graph and then extract only them as TensorFlow model, but there is currently no high-level Tensorforce function which does that, so you would need to do it yourself. I'm happy to help if you want to be in touch e.g. via Gitter, as I would be keen to add this as a general feature to the framework.

That's fine, thanks, I will have a try.

**Comment by AlexKuhnle** [comment]
I think you can leave the issue open for now, as a reminder, since I think it's something that should happen in the not-too-far future, ideally. :-)

**Comment by Vancasola** [comment]
> I think you can leave the issue open for now, as a reminder, since I think it's something that should happen in the not-too-far future, ideally. :-)

Ok, I will be very thrilled if this happens soon! :-）

**Comment by AlexKuhnle** [comment]
Will be put on roadmap, but may take a while.
